python/src/aac/plugins/validators/_validator_findings.py

```diff
@@ -18,9 +19,10 @@ class ValidatorFindings:
 
     findings: list[ValidatorFinding] = attrib(default=Factory(list), validator=validators.instance_of(list))
 
-    def add_findings(self, new_findings: list[ValidatorFinding]) -> None:
+    def add_findings(self, new_findings: Union[list[ValidatorFinding], "ValidatorFindings"]) -> None:
         """Add the new findings to the collection of all findings."""
-        self.findings.extend(new_findings)
+        findings = new_findings if isinstance(new_findings, list) else new_findings.get_all_findings()
```

Comment:
Like the additional sanity check of the `isinstance` for verifying that what's passed in truly is a list as desired. 

And the else gets you a list if you pass what would be a higher object that holds the findings, if I am following correctly?

Reply:
Given that we're claiming to accept the two types of objects described in [this comment](https://github.com/DevOps-MBSE/AaC/pull/667#discussion_r1320466604) the expectation is that if we get a `list` we just use that whereas if we get anything else we assume it's a `ValidatorFindings` object which exposes a `get_all_findings()` method. As you allude to, passing in anything else would break things.

Reply:
Would it be useful to wrap that else in a try/catch since if it is something other than the two types of findings objects that `get_all_findings()` wouldn't exist and we could then handle the break more gracefully? Or is there a higher level handling of that break that I missed/am unaware of?

Reply:
It will probably slow some things down since the checks would happen at runtime but if we want to do this, in general, I'd prefer to use [`typeguard`](https://typeguard.readthedocs.io/en/latest/userguide.html) over polluting the project with a bunch of `try`/`catch` blocks. We actually already use the `check_type` function from the `typeguard` library in a couple places, so it shouldn't be crazy to add it but I don't know the performance penalty we'll get by adding that everywhere (and it probably makes sense to do it everywhere or in very few places/nowhere, IMO).

Reply:
So, quick skim of the link I was about to ask what's the difference between it and `isintance` and it has a line stating it's "a beefed up version". 

The `typeguard` package has a [runtime component](https://typeguard.readthedocs.io/en/latest/userguide.html#using-the-import-hook) that I think helps address my confusion over the `Union` modification you made for `ValidatorFindings` and wouldn't make us have to use the string name reference. 

I lean towards wanting to have that additional type check because of the number of layers that seemed to be involved with validation.

Reply:
I think it'd make sense to try that but I don't think it's in scope for this story.

Reply:
Okay, I'll generate a backlog item for further evaluating `typeguard` and determining implementation needs.

Reply:
Here is the [issue](https://github.com/DevOps-MBSE/AaC/issues/673) associated with this discussion.